src/irr/asset/CBufferLoaderBIN.cpp

```diff
@@ -0,0 +1,67 @@
+#include "CBufferLoaderBIN.h"
+
+namespace irr
+{
+	namespace asset
+	{
+		asset::SAssetBundle CBufferLoaderBIN::loadAsset(io::IReadFile* _file, const asset::IAssetLoader::SAssetLoadParams& _params, asset::IAssetLoader::IAssetLoaderOverride* _override, uint32_t _hierarchyLevel)
+		{
+			if (!_file)
+				return {};
+
+			SContext ctx;
+
+			ctx.files = _file;
+			ctx.files->grab();
```

Comment:
no need to grab, you only grab if you expect to hold onto the object after the function returns.